### PR TITLE
Revised GoDoc content for clients package

### PR DESCRIPTION
**Pull request by tsconn23**
#31 (continued)
#46 

Also of note:
- All implementing types have been unexported. There is no reason any code outside of the clients package should directly access these types.

Signed-off-by: Trevor Conn <trevor_conn@dell.com>

### CODE REVIEW

**Comment by cloudxxx8** [comment]
add @weichou1229 for awareness this PR

File: clients/coredata/reading.go [hunk 1]
```diff
@@ -27,33 +28,47 @@ import (
...
+	// ReadingsForUOMLabel returns readings up to a specified limit for a given UOM label
 	ReadingsForUOMLabel(uomLabel string, limit int, ctx context.Context) ([]models.Reading, error)
+	// ReadingsForLabel returns readings up to a specified limit for a given label
 	ReadingsForLabel(label string, limit int, ctx context.Context) ([]models.Reading, error)
+	// ReadingsForType returns readings up to a specified limit of a given type
 	ReadingsForType(readingType string, limit int, ctx context.Context) ([]models.Reading, error)
+	// ReadingsForInterval returns readings up to a specified limit generated within a specific time period
 	ReadingsForInterval(start int, end int, limit int, ctx context.Context) ([]models.Reading, error)
+	// Add posts a new reading
 	Add(readiing *models.Reading, ctx context.Context) (string, error)
```
Comment (lenny-intel) [inline clients/coredata/reading.go]: Fix typo in "readiing"

File: clients/metadata/device.go [hunk 2]
```diff
@@ -26,47 +26,66 @@ import (
...
 	DevicesForProfile(profileid string, ctx context.Context) ([]models.Device, error)
+	// DevicesForProfileByName lists all devices for the specified profile name
 	DevicesForProfileByName(profileName string, ctx context.Context) ([]models.Device, error)
+	// DevicesForService lists all devices for the specified device service ID
 	DevicesForService(serviceid string, ctx context.Context) ([]models.Device, error)
+	// DevicesForServiceByName lists all devices for the specified device service name
 	DevicesForServiceByName(serviceName string, ctx context.Context) ([]models.Device, error)
+	// Update the specified device
 	Update(dev models.Device, ctx context.Context) error
+	// UpdateAdminState modifies a device's AdminState according to the specified device ID
```
Comment (lenny-intel) [inline clients/metadata/device.go]: For consistency change "according to the" to "for the" here and below

File: clients/metadata/device_service.go [hunk 3]
```diff
@@ -25,30 +25,33 @@ import (
...
-Service client for interacting with the device service section of metadata
+DeviceServiceClient defines the interface for interactions with the DeviceService endpoint on the EdgeX Foundry
+core-metadata service.
 */
 type DeviceServiceClient interface {
+	// Add a new device service
 	Add(ds *models.DeviceService, ctx context.Context) (string, error)
+	// DeviceServiceForName loads a device service for the specified name
 	DeviceServiceForName(name string, ctx context.Context) (models.DeviceService, error)
+	// UpdateLastConnected updates a device service's last connected timestamp according to the specified service ID
```
Comment (lenny-intel) [inline clients/metadata/device_service.go]: For consistency change "according to the" to "for the" here and below

**Comment by tsconn23** [comment]
@lenny-intel Thanks for the approval. Unfortunately it got dismissed by my rebase wherein I applied your recommended edits. If you would, please re-approve tomorrow and we'll get this pulled in.

**Pull request merged**